consensus/types.go

```diff
@@ -183,11 +182,11 @@ type CoveredFields struct {
 // TODO: Switch to a different constant because of using 128 bit values for the
 // currency.
 func CalculateCoinbase(height BlockHeight) Currency {
-	if Currency(height) >= InitialCoinbase-MinimumCoinbase {
-		return MinimumCoinbase * 100000
-	} else {
-		return (InitialCoinbase - Currency(height)) * 100000
+	base := InitialCoinbase - uint64(height)
+	if base < MinimumCoinbase {
+		base = MinimumCoinbase
 	}
+	return NewCurrency(base * 1e5)
```

Comment:
should probably be a unit name, like Bitcoin's satoshi and Finney. `1 << 80` is going to be the standard "siacoin," right?